design/multi-client-and-multi-endpoint-in-modular.md

```diff
@@ -0,0 +1,250 @@
+    1. To export RLC layer in Modular with  RLC as a float dependency will somehow provide extra features for Modular customers without bumping any versions.
+    1. If there’s a case where RLC layer is a breaking change and we use rename to avoid breaking in Modular layer. To export RLC layer will make it impossible to avoid that breaking in Modular.
+
+## Finalized Proposals
+
+With the above initial proposals and the above considerations, we get our finalized design for both multi-client and multi-endpoint.
+
+### Finalized Proposal in Multi-Client
+
+<!-- markdownlint-disable MD033 -->
+<table>
+  <tr>
+    <th>Subpath Exports</th>
+    <th>Source Code Structure</th>
+  </tr>
+  <tr>
+    <td>
+      <pre lang="typescript">
+Both sub classical clients exported as well as models
+@azure/foo
+
+</pre>
+<pre lang="typescript">
+ClientA
+@azure/foo/clientA
+@azure/foo/clientA/api
+@azure/foo/clientA/models
+</pre>
+<pre lang="typescript">
+ClientB
+@azure/foo/clientB
+@azure/foo/clientB/api
+@azure/foo/clientB/models
+</pre>
+</td>
+<td>
+<pre lang="typescript">
+Both sub classical clients exported as well as models
+src
+src/rest
+</pre>
+<pre lang="typescript">
+ClientA
+src/clientA
+src/clientA/api
+src/clientA/models
+</pre>
+<pre lang="typescript">
+ClientB
+src/clientB
+src/clientB/api
+src/clientB/models
+</pre>
+</td>
+  </tr>
+</table>
+<!-- markdownlint-enable MD033 -->
+
+### Finalized Proposal in Multi-Endpoint
+
+<!-- markdownlint-disable MD033 -->
+<table>
+  <tr>
+    <th>Subpath Exports</th>
+    <th>Source Code Structure</th>
+  </tr>
+  <tr>
+    <td>
+      <pre lang="typescript">
+Both sub classical clients exported as well as models
+@azure/foo
+
+</pre>
+<pre lang="typescript">
+ClientA
+@azure/foo/clientA
+@azure/foo/clientA/api
+@azure/foo/clientA/models
+</pre>
+<pre lang="typescript">
+ClientB
+@azure/foo/clientB
+@azure/foo/clientB/api
+@azure/foo/clientB/models
+</pre>
+</td>
+<td>
+<pre lang="typescript">
+Both sub classical clients exported as well as models
+src
+src/rest
+</pre>
+<pre lang="typescript">
+ClientA
+src/clientA
+src/clientA/api
+src/clientA/models
+</pre>
+<pre lang="typescript">
+ClientB
+src/clientB
+src/clientB/api
+src/clientB/models
+</pre>
+</td>
+  </tr>
+</table>
+<!-- markdownlint-enable MD033 -->
+
+The proposals look the same in both the Multi-Client and Multi-Endpoint case. But there's a difference in the `src/rest` between the Multi-Client and the Multi-Endpoint case. In the Multi-Client case, we will just have a `src/rest/index.ts` that exports everything from the `@azure-rest/service` standalone RLC. In the Multi-Endpoint case, we will just have a `src/rest/clientA/index.ts` that exports everything from the `@azure-rest/service-clientA` standalone RLC and a `src/rest/clientB/index.ts` that exports everything from the `@azure-rest/service-clientB` standalone RLC which is a multiple RLC packages maps to one Modular package scenario.
```

Comment:
Is this up-to-date? My understanding is that we will not re-export any rest api in Modular?

Reply:
It's a little odd to re-export the entire surface of a dependency, especially since the dependency is a floating version.

Reply:
I fixed it in another PR https://github.com/Azure/azure-sdk-for-js/pull/26602/files